### PR TITLE
WIP: Group UUID mapper

### PR DESCRIPTION
**Pull request by ahennr**
--- WIP 

Can be used to add uuids of groups the user is member of to the `KeycloakPrincipal`.

Herewith, the group uuids are contained in an optionalClaim of the `token` object  and can therefore be used in `baseCrudRepository` like this: '`?#{ principal?.context.token.optionalClaims.groups }`

--- WIP

### CODE REVIEW

File: shogun-lib/src/main/java/de/terrestris/shogun/lib/security/SecurityContextUtil.java [hunk 1]
```diff
@@ -80,19 +79,15 @@
      * @return
      */
     public Optional<User> getUserFromAuthentication(Authentication authentication) {
-        final Object principal = authentication.getPrincipal();
-        if (!(principal instanceof KeycloakPrincipal)) {
-            return Optional.empty();
-        }
-        // get user info from authentication object
-        String keycloakUserId = getKeycloakUserIdFromAuthentication(authentication);
+        String keycloakUserId = this.getKeycloakUserUuid(authentication);
```
Comment (LukasLohoff) [inline shogun-lib/src/main/java/de/terrestris/shogun/lib/security/SecurityContextUtil.java:82]: do we still need `getKeycloakUserIdFromAuthentication` ?

maybe we can mark it deprecated, or just replace the implementation and keep the name if it does not have any benefit over `getKeycloakUserUuid`

File: shogun-lib/src/main/java/de/terrestris/shogun/lib/security/SecurityContextUtil.java [hunk 2]
```diff
@@ -168,4 +167,24 @@ public boolean isInterceptorAdmin() {
                 StringUtils.endsWithIgnoreCase(grantedAuthority.getAuthority(), "ADMIN")
         );
     }
+
+    public String getKeycloakUserUuid(Authentication authentication) {
```
Comment (LukasLohoff) [inline shogun-lib/src/main/java/de/terrestris/shogun/lib/security/SecurityContextUtil.java:176]: some documentation would be helpful

Comment (LukasLohoff) [reply shogun-lib/src/main/java/de/terrestris/shogun/lib/security/SecurityContextUtil.java:176]: same for the two methods below

**Review by LukasLohoff — CHANGES_REQUESTED** [review]
Very nice!

Code looks good for me, I think the documentation can be improved a bit:

- do we need to adjust any Keycloak settings for this to work? 
- replacing existing methods or making clear what's the difference (see comments)

File: shogun-lib/src/main/java/de/terrestris/shogun/lib/security/SecurityContextUtil.java (hunk 1 above)
Comment (ahennr) [reply shogun-lib/src/main/java/de/terrestris/shogun/lib/security/SecurityContextUtil.java:82]: Yes it's still used in `getUserBySession`

**Review by LukasLohoff — APPROVED** [review]

**Comment by LukasLohoff** [comment]
This isn't need for the grid rework anymore correct? I'll merge this now - maybe I can continue with this sometime

the other MR is work in progress as well anyway

**Pull request merged**